python/pyiceberg/catalog/rest.py

```diff
@@ -0,0 +1,435 @@
+        """
+        self.host = host
+        if client_id and client_secret:
+            self.token = self._fetch_access_token(client_id, client_secret)
+        elif token:
+            self.token = token
+        else:
+            raise ValueError("Either set the client_id and client_secret, or provide a valid token")
+        self.config = self._fetch_config(properties)
+        super().__init__(name, properties)
+
+    @staticmethod
+    def _split_credential(token: str) -> Tuple[str, str]:
+        """Splits the token in a client id and secret
+
+        Args:
+            token: The token with a semicolon as a separator
+
+        Returns:
+            The client id and secret
+        """
+        client, secret = token.split(":")
+        return client, secret
+
+    @property
+    def headers(self) -> Properties:
+        return {
+            AUTHORIZATION_HEADER: f"{BEARER_PREFIX} {self.token}",
+            "Content-type": "application/json",
+            "X-Client-Version": __version__,
+        }
+
+    def url(self, endpoint: str, prefixed: bool = True, **kwargs) -> str:
+        """Constructs the endpoint
+
+        Args:
+            prefixed: If the prefix return by the config needs to be appended
+
+        Returns:
+            The base url of the rest catalog
+        """
+
+        url = self.host
+        url = url + "v1/" if url.endswith("/") else url + "/v1/"
+
+        if prefixed:
+            url += self.config.get(PREFIX, "")
+            url = url if url.endswith("/") else url + "/"
+
+        return url + endpoint.format(**kwargs)
+
+    def _fetch_access_token(self, client_id: str, client_secret: str) -> str:
+        data = {GRANT_TYPE: CLIENT_CREDENTIALS, CLIENT_ID: client_id, CLIENT_SECRET: client_secret, SCOPE: CATALOG_SCOPE}
+        url = self.url(Endpoints.get_token, prefixed=False)
+        # Uses application/x-www-form-urlencoded by default
+        response = requests.post(url=url, data=data)
+        try:
+            response.raise_for_status()
+        except HTTPError as exc:
+            self._handle_non_200_response(exc, {401: BadCredentialsError})
+
+        return TokenResponse(**response.json()).access_token
+
+    def _fetch_config(self, properties: Properties) -> Properties:
+        response = requests.get(self.url(Endpoints.get_config, prefixed=False), headers=self.headers)
+        response.raise_for_status()
+        config_response = ConfigResponse(**response.json())
+        config = config_response.defaults
+        config.update(properties)
+        config.update(config_response.overrides)
+        return config
+
+    def _split_identifier_for_path(self, identifier: Union[str, Identifier]) -> Properties:
+        identifier = self.identifier_to_tuple(identifier)
+        return {"namespace": NAMESPACE_SEPARATOR.join(identifier[:-1]), "table": identifier[-1]}
+
+    def _split_identifier_for_json(self, identifier: Union[str, Identifier]) -> Dict[str, Union[Tuple[str, ...], str]]:
```

Comment:
Nit: The return type could be `Dict[str, Union[str, Identifier]]` instead of using `Tuple[str, ...]`.

Reply:
Nice one, thanks!